ts_src/psbt.ts

```diff
@@ -733,59 +765,57 @@ export class Psbt {
     });
 
     // generate output blinding factors
-    const numOutputs = this.data.outputs.length - 1; // exclude fees output
-    const outputAbfs = this.data.outputs
-      .slice(0, numOutputs)
-      .map(() => randomBytes(opts!));
-    const outputVbfs = this.data.outputs
-      .slice(0, numOutputs - 1)
-      .map(() => randomBytes(opts!));
+    const numOutputs = outputsIndexToBlind.length;
+    const outputAbfs = range(numOutputs).map(() => randomBytes(opts!));
+    const outputVbfs = range(numOutputs - 1).map(() => randomBytes(opts!));
+
     const finalVbf = confidential.valueBlindingFactor(
       inputValues,
-      outputValues,
+      outputValues.filter((_, index) => outputsIndexToBlind!.includes(index)),
```

Comment:
can you explain this?

Reply:
Better wrap in in variable to make it more readble and to comment in it, I agree